pkg/scheduler/objects/queue.go

```diff
@@ -1255,3 +1299,75 @@ func (sq *Queue) String() string {
 	return fmt.Sprintf("{QueuePath: %s, State: %s, StateTime: %x, MaxResource: %s}",
 		sq.QueuePath, sq.stateMachine.Current(), sq.stateTime, sq.maxResource)
 }
+
+// incRunningApps increments the number of running applications for this queue (recursively).
+// Guarded against going over the max set. Combined with the decRunningApps guard against below zero
+// this guard should allow self-heal.
+func (sq *Queue) incRunningApps(appID string) {
+	if sq == nil {
+		return
+	}
+	if sq.parent != nil {
+		sq.parent.incRunningApps(appID)
+	}
+	sq.Lock()
+	defer sq.Unlock()
+	delete(sq.allocatingAcceptedApps, appID)
```

Comment:
We need to be sure that when an application goes to a terminal state that we also remove from sq.allocatingAcceptedApps, otherwise we potentially leak slots if some placeholders exist but we kill the app.